python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -239,3 +245,250 @@ def readFits(cls, filename):
+        dtype=float,
+        doc="Sigma for outlier rejection; ignored if stacking_statistic != 'MEANCLIP'.",
+        default=4
+    )
+    num_iter = pexConfig.Field(
+        dtype=int,
+        doc="Number of iterations of outlier rejection; ignored if atackingStatistic != 'MEANCLIP'.",
+        default=3
+    )
+    bad_mask_planes = pexConfig.ListField(
+        dtype=str,
+        doc="Mask planes that, if set, lead to associated pixels not being included in the stacking of the "
+            "bright star stamps.",
+        default=('BAD', 'CR', 'CROSSTALK', 'EDGE', 'NO_DATA', 'SAT', 'SUSPECT', 'UNMASKEDNAN')
+    )
+    do_mag_cut = pexConfig.Field(
+        dtype=bool,
+        doc="Apply magnitude cut before stacking?",
+        default=False
+    )
+    mag_limit = pexConfig.Field(
+        dtype=float,
+        doc="Magnitude limit, in Gaia G; all stars brighter than this value will be stacked",
+        default=18
+    )
+
+
+class StackBrightStarsTask(pipeBase.CmdLineTask):
+    """Stack bright stars together to build an extended PSF model.
+    """
+    ConfigClass = StackBrightStarsConfig
+    _DefaultName = "stack_bright_stars"
+
+    def __init__(self, initInputs=None, *args, **kwargs):
+        pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+
+    def _set_up_stacking(self, example_stamp):
+        """Configure stacking statistic and control from config fields.
+        """
+        stats_control = afwMath.StatisticsControl()
+        stats_control.setNumSigmaClip(self.config.num_sigma_clip)
+        stats_control.setNumIter(self.config.num_iter)
+        bad_masks = self.config.bad_mask_planes
+        if bad_masks:
+            and_mask = example_stamp.mask.getPlaneBitMask(bad_masks[0])
+            for bm in bad_masks[1:]:
+                and_mask = and_mask | example_stamp.mask.getPlaneBitMask(bm)
+            stats_control.setAndMask(and_mask)
+        stats_flags = afwMath.stringToStatisticsProperty(self.config.stacking_statistic)
+        return stats_control, stats_flags
+
+    def run(self, bss_ref_list, region_name=None):
+        """Read input bright star stamps and stack them together.
+
+        The stacking is done iteratively over smaller areas of the final model
+        image to allow for a great number of bright star stamps to be used.
+
+        Parameters
+        ----------
+        bss_ref_list : `list` of
+                `lsst.daf.butler._deferredDatasetHandle.DeferredDatasetHandle`
+            List of available bright star stamps data references.
+        region_name : `str`, optional
+            Name of the focal plane region, if applicable. Only used for
+            logging purposes, when running over multiple such regions
+            (typically from `MeasureExtendedPsfTask`)
+        """
+        log_message = f'Building extended PSF from stamps extracted from {len(bss_ref_list)} detector images '
+        if region_name:
+            log_message += f'for region "{region_name}".'
```

Comment:
Maybe it would be cleaner to move the extra space at the end of L337 to the beginning of the f-string in L339.